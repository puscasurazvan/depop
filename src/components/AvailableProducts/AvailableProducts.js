import React, { useEffect } from 'react'
import Product from '../Product'

import PropTypes from 'prop-types'

const AvailableProducts = ({ products, setCount }) => {
  const filteredProducts = products.filter(({ sold }) => sold === true)

  useEffect(() => {
    if (filteredProducts) {
      setCount(filteredProducts.length)
    }
  }, [setCount, filteredProducts])

  return filteredProducts.map(
    ({ id, title, brand, size, price, img, sold }) => {
      return (
        <Product
          key={id}
          productTitle={title}
          imageURL={img}
          productBrand={brand}
          productSize={size}
          productPrice={price}
        />
      )
    }
  )
}

AvailableProducts.propTypes = {
  products: PropTypes.array.isRequired,
  setCount: PropTypes.func,
}

export default AvailableProducts
