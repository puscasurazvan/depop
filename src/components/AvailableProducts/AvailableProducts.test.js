import React from 'react'
import { mount } from 'enzyme'
import AvailableProducts from './AvailableProducts'

const minProps = {
  products: [],
  setCount: jest.fn(),
}

describe('AvailableProducts', () => {
  it('should render with the right props', () => {
    const wrapper = mount(<AvailableProducts {...minProps} />)
    expect(wrapper.props().products).toStrictEqual([])
    expect(wrapper.find('AvailableProducts').length).toEqual(1)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<AvailableProducts {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
