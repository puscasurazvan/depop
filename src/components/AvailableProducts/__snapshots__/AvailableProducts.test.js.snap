// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailableProducts should render without exploding 1`] = `ReactWrapper {}`;
